### PR TITLE
Blocks: fix translation for Product Rating blocks

### PR DESCRIPTION
**Pull request by gigitux**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR belongs to [50048](https://github.com/woocommerce/woocommerce/issues/50048). All the PRs will be merged into the `fix/translation` branch, and after that, this latter will be merged in the trunk.

### Issue

This PR addresses an issue where translations don't work for blocks that are inside the `/atomic/blocks` path. 

Refer to a similar issue for more context: [Issue #6737](https://github.com/woocommerce/woocommerce-blocks/pull/6737).

#### Solution

Based on the discussion in [Issue #42646](https://github.com/woocommerce/woocommerce/issues/42646), the "atomic" folder structure shoud be removed, and all blocks will be consolidated into the `blocks` folder. This PR implements the following changes:

- Moves utility functions to the `blocks` folder
- Relocates the Product Rating block to the `blocks` folder

These changes ensure that the blocks are properly registered on the server side, resolving the translation issue.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout trunk
2. Switch to the Italian or Spanish language (or some language that has 100% of string translated)
3. Visit `wp-admin/site-editor.php?postId=woocommerce%2Fwoocommerce%2F%2Fsingle-product&postType=wp_template&canvas=edit`
4. Check the Product Rating
5. Ensure that the block isn't translated
6. Checkout this branch
7. Repeat 2-4
8. Ensure that the block is translated

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Blocks: fix translation for Product Rating blocks

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-rating/index.ts [hunk 1]
```diff
@@ -20,9 +20,6 @@ const blockConfig: BlockConfiguration = {
 	edit,
 };
 
-registerBlockSingleProductTemplate( {
-	blockName: 'woocommerce/product-rating',
-	blockMetadata: metadata,
-	blockSettings: blockConfig,
-	isAvailableOnPostEditor: true,
-} );
+console.log( 'ci entro?' );
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/blocks/product-rating/index.ts]: Sì! non ce n'è più bisogno!

```suggestion
```

File: plugins/woocommerce-blocks/assets/js/blocks/shared/with-product-selector.js
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/blocks/shared/with-product-selector.js:1]: I copied pasted this file

File: plugins/woocommerce-blocks/assets/js/blocks/shared/use-is-descendent-of-single-product-template.tsx
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/blocks/shared/use-is-descendent-of-single-product-template.tsx:1]: I copied pasted this file

File: plugins/woocommerce-blocks/assets/js/blocks/shared/use-is-descendent-of-single-product-block.tsx
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/blocks/shared/use-is-descendent-of-single-product-block.tsx:1]: I just copied pasted this file

File: plugins/woocommerce-blocks/assets/js/blocks/shared/product-selector.tsx
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/blocks/shared/product-selector.tsx:1]: I just copied pasted this file

File: plugins/woocommerce-blocks/assets/js/blocks/shared/editor.scss
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/blocks/shared/editor.scss:1]: I just copied pasted this file

File: plugins/woocommerce-blocks/assets/js/blocks/shared/config.tsx
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/blocks/shared/config.tsx:1]: I just copied pasted this file

File: plugins/woocommerce-blocks/assets/js/blocks/product-rating/index.ts (hunk 1 above)
Comment (gigitux) [reply plugins/woocommerce-blocks/assets/js/blocks/product-rating/index.ts]: Thanks for catching this! Fixed with 299f55f2466a369dfb3a6799eb60379d89c7fb63 :D

**Comment by gigitux** [comment]
I'm closing this PR because the culprit was `registerBlockSingleProductTemplate` function. https://github.com/woocommerce/woocommerce/pull/50615 fixes the issue!

**Pull request closed without merging**